### PR TITLE
migration draft

### PR DESCRIPTION
**Pull request by Shenniey**
Issue #

### CODE REVIEW

File: Nodejs/Product/Nodejs/NodejsPackage.cs [hunk 1]
```diff
@@ -94,11 +94,16 @@ protected override void Initialize()
             RegisterProjectFactory(new NodejsProjectFactory(this));
 
             // Add our command handlers for menu (commands must exist in the .vsct file)
-            var commands = new List<Command> {
-                new ImportWizardCommand(),
-            };            
+            var importWizardCmd = new List<Command> {
```
Comment (joj) [inline Nodejs/Product/Nodejs/NodejsPackage.cs:97]: This is not wrong, but I would prefer migration to be in its own package, so we can easily remove it when it's no longer needed (migration should be a temporary thing, I believe)

File: Nodejs/Product/Nodejs/NodejsTools.vsct [hunk 2]
```diff
@@ -59,6 +59,10 @@
         <Parent guid="guidNodeToolsWseContextMenuCmdSet" id="idmNpm"/>
       </Group>
 
+      <Group guid="guidMigrateToJspsCmdSet" id="cmdidJspsMigrate">
```
Comment (joj) [inline Nodejs/Product/Nodejs/NodejsTools.vsct:62]: Same with VSCT, the command code, etc. It should go in a new Migration vsix or something, completely independent of the rest of the things, if at all possible.

File: Nodejs/MigrateToJsps/Jsps/JspsProjectCreator.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        public JspsProjectCreator(string njsprojDir, NjsprojFileModel njsprojFileModel, string newProjectParentDir)
+        {
+            this.njsprojDir = njsprojDir;
+            this.njsprojFileModel = njsprojFileModel;
+            this.newProjectParentDir = newProjectParentDir;
+        }
+
+        public void CreateJspsProject() // to be called by outsiders
+        {
+            var parentDir = Path.Combine(newProjectParentDir, njsprojFileModel.ProjectName + "-migrated");
```
Comment (joj) [inline Nodejs/MigrateToJsps/Jsps/JspsProjectCreator.cs:25]: The other way around: NTVS gets "-old", jsps gets the old name :)

Comment (Shenniey) [reply Nodejs/MigrateToJsps/Jsps/JspsProjectCreator.cs]: done!

File: Nodejs/Product/Nodejs/NodejsPackage.cs (hunk 1 above)
Comment (Shenniey) [reply Nodejs/Product/Nodejs/NodejsPackage.cs:98]: i'm still working on this -- will update this comment once that is done

File: Nodejs/Product/NodejsToolsVsix/NodejsToolsVsix.csproj [hunk 4]
```diff
@@ -71,6 +71,14 @@
       <VSIXSubPath>.</VSIXSubPath>
       <Private>True</Private>
     </ProjectReference>
+    <ProjectReference Include="..\..\MigrateToJsps\MigrateToJsps.csproj">
+      <Project>{FCAE2282-8ADC-46C8-B4C8-C25A76BD9567}</Project>
+      <Name>MigrateToJsps</Name>
+      <IncludeOutputGroupsInVSIX>BuiltProjectOutputGroup%3bBuiltProjectOutputGroupDependencies%3bGetCopyToOutputDirectoryItems%3bSatelliteDllsProjectOutputGroup%3b</IncludeOutputGroupsInVSIX>
```
Comment (joj) [inline Nodejs/Product/NodejsToolsVsix/NodejsToolsVsix.csproj:77]: %3b should be ;

File: Nodejs/Product/Nodejs/NodejsPackage.cs (hunk 1 above)
Comment (Shenniey) [reply Nodejs/Product/Nodejs/NodejsPackage.cs:98]: Resolved offline -- decided against moving the migration into its own package, but will refactor to keep migration logic as easily separable as possible from the rest of nodejstools (i.e. move migrate command to its own .vsct file).

File: Nodejs/Product/Nodejs/NodejsTools.vsct (hunk 2 above)
Comment (Shenniey) [reply Nodejs/Product/Nodejs/NodejsTools.vsct:62]: Resolved offline -- decided to keep cmd in the same vsix, but will move migrate command to its own .vsct file (see [above comment](https://github.com/microsoft/nodejstools/pull/2452#discussion_r1137853978)).

**Comment by Shenniey** [comment]
MOVING TO #2456 -- all relevant comments applied to new PR.

File: Nodejs/Product/Nodejs/NodejsPackage.cs (hunk 1 above)
Comment (Shenniey) [reply Nodejs/Product/Nodejs/NodejsPackage.cs:98]: Moved to item #&#x2060;1 in #2456

File: Nodejs/Product/NodejsToolsVsix/NodejsToolsVsix.csproj (hunk 4 above)
Comment (Shenniey) [reply Nodejs/Product/NodejsToolsVsix/NodejsToolsVsix.csproj:77]: Moved to item #&#x2060;2 in #2456

**Pull request closed without merging**